### PR TITLE
Rename icu filters to make it easier to generate icudt.dat and icudt_*.dat files

### PR DESCRIPTION
**Pull request by steveisok**

### CODE REVIEW

**Review by EgorBo — APPROVED** [review]

File: eng/icu.proj [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <!-- Only building for wasm is supported -->
   <Target Name="Build">
     <ItemGroup>
-      <IcuFilter Include="optimal;optimal_no_CJK;EFIGS;CJK" />
+      <IcuFilter Include="icudt;icudt_optimal_no_CJK;icudt_EFIGS;icudt_CJK" />
```
Comment (akoeplinger) [inline eng/icu.proj]: nit, maybe rename icudt_optimal_no_CJK -> icudt_no_CJK?

**Comment by naricc** [comment]
@DrewScoggins This should unbreak BDN.

**Review by akoeplinger — APPROVED** [review]

**Review by safern — COMMENTED** [review]
LGTM

**Comment by DrewScoggins** [comment]
Is the plan to let this flow with Maestro, or port this over manually like we did with https://github.com/dotnet/icu/pull/20?

**Pull request merged**